install.sh

```diff
@@ -454,7 +454,16 @@ do_install() {
 				fi
 				$sh_c "$pkg_manager install -y -q $pre_reqs"
 				$sh_c "$config_manager --add-repo $yum_repo"
-
+                if [ -n "$mirror" ]; then
+                    case "$mirror" in
+                        Aliyun)
+                            sed -i "s#download.docker.com#mirrors.aliyun.com/docker-ce#g" /etc/yum.repos.d/docker-ce.repo
+                            ;;
+                        AzureChinaCloud)
+                            sed -i "s#download.docker.com#mirror.azure.cn/docker-ce#g" /etc/yum.repos.d/docker-ce.repo
+                            ;;
+                    esac
+                fi
```

Comment:
Looks like the indentation is using spaces instead of tabs here.

Also wondering if we should repeat the `case` here, or if we should (e.g.) add a `RPM_REPO_URL` variable to the existing code higher up; https://github.com/docker/docker-install/blob/6580690119f66d3e31b8bc84e7ac9f9e3d68559b/install.sh#L63-L70 (or even re-use `DOWNLOAD_URL` to always update the `docker-ce.repo` URL?

And if that variable is set to do the replace.

Reply:
More in general though, I'm a bit unsure if we should keep these mirror options; are they still useful? I recall that support for our CDN's has improved behind the GFW, so perhaps it's no longer needed for "most" situations.

Looking at the mirrors, is the aliyun mirror still operational?

- https://mirrors.aliyun.com/docker-ce redirects to http://pub.mirrors.aliyun.com/docker-ce/ (does not resolve for me)
- https://mirror.azure.cn/docker-ce (looks to be working still)

There's also the rootless-install script; which has the URLs hard-coded; https://github.com/docker/docker-install/blob/6580690119f66d3e31b8bc84e7ac9f9e3d68559b/rootless-install.sh

Which adds a bit to my concern if adding extra complexity is worth it, or if we should remove it altogether.

We already have the `DOWNLOAD_URL` variable (and/or `RPM_REPO_URL`); if we added the replace for `docker-ce.repo`, we could remove the switch for these specific mirrors


Will have to check with others what's the desired approach